R/pmcmc.R

```diff
@@ -66,6 +65,59 @@ pmcmc <- function(pars, filter, initial = NULL, control = NULL) {
 }
 
 
+##' Run a pMCMC, with sensible random number behaviour, but schedule
+##' execution of the chains yourself. Use this if you want to
+##' distribute chains over (say) the nodes of an HPC system.
+##'
+##' @title pMCMC with manual chain scheduling
+##'
+##' @inheritParams pmcmc
+##' @export
+pmcmc_chains_prepare <- function(pars, filter, initial = NULL, control = NULL) {
+  assert_is(pars, c("pmcmc_parameters", "pmcmc_parameters_nested"))
+  assert_is(filter, "particle_filter")
+  assert_is(control, "pmcmc_control")
+
+  if (control$n_workers != 1) {
+    stop("'n_workers' must be 1")
+  }
+  if (!control$use_parallel_seed) {
+    stop("'use_parallel_seed' must be TRUE")
```

Comment:
Can we just auto-set this given it wouldn't break anything if it was unintentionally set to TRUE? (I don't think it would?)

Reply:
I'm not a fan of changing options behind the user's back because then you get odd dependencies between options. We might change the default at some point